packages/andrjs/src/clients/InjectiveClient.ts

```diff
@@ -239,43 +238,14 @@ export default class InjectiveClient
   async broadcast(
     tx: InjTxRaw,
     timeoutMs = 60000,
-    pollIntervalMs = 3000
   ): Promise<DeliverTxResponse> {
-    const resp = await this.signingClient!.broadcastBlock(tx);
-    //This code is duplicated from the CosmWasmClient to correct return types
-    let timedOut = false;
-    const txPollTimeout = setTimeout(() => {
-      timedOut = true;
-    }, timeoutMs);
-
-    const pollTx: (
-      txId: string
-    ) => ReturnType<ChainClient["broadcast"]> = async (txId) => {
-      if (timedOut)
-        throw new Error(
-          `Transaction with ID ${txId} was submitted but was not yet found on the chain. You might want to check later. There was a wait of ${
-            timeoutMs / 1000
-          } seconds`
-        );
-      await sleep(pollIntervalMs);
-      const result = await this.queryClient!.getTx(txId);
-      return result
-        ? ({ ...result, transactionHash: result.hash } as DeliverTxResponse)
-        : pollTx(txId);
-    };
-
-    return new Promise((resolve, reject) =>
-      pollTx(resp.txHash).then(
-        (val) => {
-          clearTimeout(txPollTimeout);
-          resolve(val);
-        },
-        (err) => {
-          clearTimeout(txPollTimeout);
-          reject(err);
-        }
-      )
-    );
+    const resp = await this.signingClient!.broadcastBlock(tx, 3);
```

Comment:
Has this method been updated on the cosmjs side to use ASYNC instead of BLOCK?

@arun567

Reply:
Yes, The broadcasting modes are updated in the latest cosmjs, Previuosly there were 4 modes including BLOCK and ASYNC, and Now 2 of them are deprecated and BLOCK is among the deprecated.